atlascope/core/fields/cube.py

```diff
@@ -0,0 +1,56 @@
+from collections import UserList
+import re
+from typing import Iterable
+
+from django.db.models import Field, FloatField, Func
+from psycopg2.extensions import AsIs
+
+from atlascope.core.utils import register_adapter
+
+__all__ = ['Cube', 'CubeDistance', 'CubeField']
+
+
+class Cube(UserList):
```

Comment:
Could you explain the purpose of this addition? It seems like this is the data structure we use to store a fingerprint, but why is it a Cube?

Reply:
Great question. We define `Cube` so that [psycopg2](https://www.psycopg.org/docs/) can convert it to appropriate SQL. [This function](https://github.com/atlascope/atlascope/pull/191/files#diff-db5454fe2adc4a5225b50335b0c1f73e85f54a1383549418e98e93ac8caff21cR17-R20) defines that behavior. The SQL translation would be incorrect for just a plain list.